packages/datetime/src/dateInput.tsx

```diff
@@ -478,8 +478,13 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         this.safeInvokeInputProp("onKeyDown", e);
     };
 
-    // blur DOM event listener (not React event)
-    private handlePopoverBlur = () => this.handleClosePopover();
+    // keyboard DOM event listener (not React event)
+    private handlePopoverBlur = (e: KeyboardEvent) => {
+        if (e.which === Keys.TAB && !e.shiftKey) {
+            e.target.dispatchEvent(new FocusEvent("blur"));
```

Comment:
why? this is suspicious. we're not in the habit of dispatching DOM events from blueprint code (except where absolutely necessary in tests).